JoonasHeinonen/escape-from-zordoom-prison
# Feature/door

### [Opening / closing doors](https://trello.com/c/Pd1OqKXK/70-create-a-door-and-making-opening-and-closing-animation)

https://user-images.githubusercontent.com/33086486/226181200-0c6ac388-dafa-4c54-9fa6-2f4f7842eaf0.mp4

**Changes:**
 - Added the `TODO Manager` plugin into the project.
 - Created the initial enemy placeholder with the scene and the script.
 - Enemies can die and take damage, added animations.
 - Created the door with the animation, logic and necessary effects.

**How to test:**
 - See that the doors and the door smoke effects work properly.
 - See that the code is readable, good.
 - Ensure that the TODO tags can be reviewed from the bottom bar by clicking the `TODO` -tab

**Sources cited:**
 - [TODO manager](https://github.com/OrigamiDev-Pete/TODO_Manager)
 - [AnimationNodeStateMachinePlayback](https://docs.godotengine.org/en/stable/classes/class_animationnodestatemachineplayback.html#animationnodestatemachineplayback)
 - [AnimationTree](https://docs.godotengine.org/en/stable/classes/class_animationtree.html#animationtree)
